### ISSUE TITLE
[1.x] Question about mvideoroom demo javascript code

### ISSUE THREAD

**Issue by ioudas** [meetecho/janus-gateway#2929]
Hello,
I have a question about this line:  https://github.com/meetecho/janus-gateway/blob/03db74a81f6ab294cd3164211a985ed2a11dae37/html/mvideoroomtest.js#L594

1. The `remoteFeed` variable is set in `success` callback.
2. The `creatingFeed` variable is set in `attached` event handler

Since 1. always comes before 2. the line `if(creatingFeed)` doesn't seem to me like it serves it's intended purpose. When the handle is created but `attached` event has not arrived yet the code will send another `subscribe` anyway. 

Which undesirable scenario is this code preventing?
Thank you

**Comment by lminiero** [comment]
It's meant to avoid we create the handle more than once. In that demo, we use a single handle/PeerConnection to receive all other participants, and so the first time we have to create it, for other participants joining/leaving it just needs to be updated.

**Comment by ioudas** [comment]
Thank you for reply, but I understand that intention. I didn't express myself clearly. 
Let me rephrase the question: Why is `creatingFeed` set in 'attached' event handler and not in `success` callback of `attach()` method?

`success` always comes before `attached` therefore I don't think the `creatingFeed` condition can ever be executed (because handle will never be null if `creatingFeed` is still true.

**Comment by lminiero** [comment]
Because the `success` only tells you if we managed to attach to the plugin, while what we want to know is whether we managed to create a subscription (which involves further communication on that handle). As such, until we've managed to create a full subscription the first time, we prevent any further update.

**Comment by ioudas** [comment]
Yes, exactly, but in `success` the handle (`remoteFeed`) is set, therefore we will never get to the `if(creatingFeed )` after that.

If the intent is to wait until attached, then either conditions should be reversed - `if(creatingFeed )`  above `if(remoteFeed)`  or remoteFeed must be set in `attached` handler.
I think former is more desirable (makes more sense to set handle the moment we got it).

**Comment by lminiero** [comment]
Ah ok, I think I understand your point now. You mean we should simply invert the checks, since `creatingFeed` is what gets the ball rolling, and the previous check may cause requests to be sent ahead of time. I do think it might make sense, especially in case of race conditions at startup (someone joins while you're creating the first subscription). Pinging @atoppi so he can have a look too.

**Comment by ioudas** [comment]
Yeah I should have phrased it this way in the first message :D

Thank for confirming the intent, I'm currently implementing the JS client in our project and I'll do it the same way (waiting for 'attached')

**Comment by atoppi** [comment]
> If the intent is to wait until attached, then either the conditions should be reversed - if(creatingFeed ) above if(remoteFeed)

I agree.

![image](https://user-images.githubusercontent.com/3405061/160376554-43b314a3-66cb-4c36-851b-eccc52971baa.png)

**Comment by lminiero** [comment]
This comment should address it: I also modified the comments a bit to make the whole process easier to understand. Hopefully it should be less confusing now (for you and me both :smile: ). Thanks for spotting this!
